File: lib/domain/entities/map_pym/batiment.dart
import 'package:flutter/foundation.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'batiment.freezed.dart';

@freezed
abstract class Batiment with _$Batiment {
  const factory Batiment({
    @required int id,
    @nullable @required String nom,
    @nullable @required int nbEtage,
    @nullable @required String description,
    @nullable @required bool accesHandicape,
    @nullable @required String url,
    @nullable @required String adresse,
    @nullable @required double latitude,
    @nullable @required double longitude,
    @nullable @required bool isVisibleAR,
    @nullable @required String img_url,
  }) = _Batiment;
}

File: unity/App_pym_AR/Assets/Scripts/Batiment.cs
using System;
using System.Collections.Generic;

[Serializable]
public class Batiment
{
    public int id;
    public string nom;
    public int nbEtage;
    public string description;
    public bool accesHandicape;
    public string url;
    public string adresse;
    public float latitude;
    public float longitude;
    public bool isVisibleAR;
    public List<Entreprise> entreprises;
}
